From 234fc818eb47ed6dd6333452dfcbe081bca3bb89 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:53:19 -0400
Subject: [tools] perf/trace: Add fallback definition of EFD_SEMAPHORE

Message-id: <1408953210-26343-312-git-send-email-jolsa@redhat.com>
Patchwork-id: 88944
O-Subject: [PATCH RHEL7.1 BZ1133083 311/322] perf trace: Add fallback definition of EFD_SEMAPHORE
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 79d26a6a19ace19faabf8d8d27d3430be2e26d34
Author: Ben Hutchings <ben@decadent.org.uk>
Date: Thu Feb 6 01:00:35 2014 +0000

description
===========
glibc 2.17 is missing this on sparc, despite the fact that it's not
architecture-specific.

Fixes: 49af9e93adfa ('perf trace: Beautify eventfd2 'flags' arg')
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index 896f270..619d11c 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -37,6 +37,10 @@
 # define MADV_UNMERGEABLE 13
 #endif
 
+#ifndef EFD_SEMAPHORE
+# define EFD_SEMAPHORE  1
+#endif
+
 struct tp_field {
  int offset;
  union {
-- 
1.7.1